Content.Shared/Labels/EntitySystems/SharedHandLabelerSystem.cs

```diff
@@ -57,64 +57,92 @@ protected virtual void UpdateUI(Entity<HandLabelerComponent> ent)
     {
     }
 
-    private void AddLabelTo(EntityUid uid, HandLabelerComponent? handLabeler, EntityUid target, out string? result)
+    private void AddLabelTo(EntityUid uid, EntityUid User, HandLabelerComponent? handLabeler, EntityUid target)
     {
         if (!Resolve(uid, ref handLabeler))
         {
-            result = null;
             return;
         }
 
         if (handLabeler.AssignedLabel == string.Empty)
         {
-            if (_netManager.IsServer)
-                _labelSystem.Label(target, null);
-            result = Loc.GetString("hand-labeler-successfully-removed");
+            RemoveLabelFrom(uid, User, handLabeler, target);
             return;
         }
+
         if (_netManager.IsServer)
             _labelSystem.Label(target, handLabeler.AssignedLabel);
-        result = Loc.GetString("hand-labeler-successfully-applied");
+
+        _popupSystem.PopupClient(Loc.GetString("hand-labeler-successfully-applied"), User, User);
+
+        // Log labeling
+        _adminLogger.Add(LogType.Action, LogImpact.Low,
+            $"{ToPrettyString(User):user} labeled {ToPrettyString(target):target} with {ToPrettyString(uid):labeler}");
+    }
+
+    private void RemoveLabelFrom(EntityUid uid, EntityUid User, HandLabelerComponent? handLabeler, EntityUid target)
```

Comment:
Same as above, however `handLabeler` is also never used in this method all, so instead it can just be removed